src/refiners/foundationals/latent_diffusion/schedulers/euler.py

```diff
@@ -22,31 +24,45 @@ def __init__(
             device=device,
             dtype=dtype,
         )
-        self.sigmas = self.noise_std / self.cumulative_scale_factors
-        self.sigmas[0] = 0.0
+        self.sigmas = self._generate_sigmas().to(device=self.device, dtype=self.dtype)
 
     def _generate_timesteps(self) -> Tensor:
+        # using "trailing" timestep
```

Comment:
Why using "trailing" specifically?

Reply:
I was just testing if others timespaces worked better. I will change it to "leading" as in the DDIM implementation.